### PR TITLE
Add Apple Developer docs extension

### PR DESCRIPTION
**Pull request by CaliCastle**
## Description

This extension helps you quickly search through the Apple Developers documentation site.

(Action includes `Open in Browser`, `Copy URL` and `Copy URL in Markdown`)

## Screencast

https://user-images.githubusercontent.com/10830749/180649688-7a365d7c-a3f5-4a9e-825b-72d9b4ef92bb.mp4

https://user-images.githubusercontent.com/10830749/180649697-b3bdcf60-d3f3-452a-a70d-80fd2fc7e147.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by thomaspaulmann** [comment]
This is a great addition @CaliCastle and congrats to your first contribution. We've just released a set of helpers that make these kind of extensions much simpler to build. You can use the `useFetch` hook to fetch data from the network. The hook caches requests, handles cancellation and allows you to easier build your extension. Switching to the hook would greatly improve the experience of the extension. Mainly it will be faster due to caching. Would you mind using the hook? You can follow [this commit](https://github.com/raycast/extensions/commit/b8cc8be9c0b43439940ec008af1b9885ae33d110).

**Comment by CaliCastle** [comment]
> This is a great addition @CaliCastle and congrats to your first contribution. We've just released a set of helpers that make these kind of extensions much simpler to build. You can use the `useFetch` hook to fetch data from the network. The hook caches requests, handles cancellation and allows you to easier build your extension. Switching to the hook would greatly improve the experience of the extension. Mainly it will be faster due to caching. Would you mind using the hook? You can follow [this commit](https://github.com/raycast/extensions/commit/b8cc8be9c0b43439940ec008af1b9885ae33d110).

Sounds good Thomas! Thanks for providing this info! I'm going to refactor it right away

**Comment by CaliCastle** [comment]
> This is a great addition @CaliCastle and congrats to your first contribution. We've just released a set of helpers that make these kind of extensions much simpler to build. You can use the `useFetch` hook to fetch data from the network. The hook caches requests, handles cancellation and allows you to easier build your extension. Switching to the hook would greatly improve the experience of the extension. Mainly it will be faster due to caching. Would you mind using the hook? You can follow [this commit](https://github.com/raycast/extensions/commit/b8cc8be9c0b43439940ec008af1b9885ae33d110).

Hi Thomas, I've refactored using `useFetch`

**Comment by thomaspaulmann** [comment]
Works as expected and the caching makes it a lot snappier. Great work. A few suggestions that you can tackle in a follow up PR:
- Show the accessories with icons instead of text. This would improve the layout. You can use the icons that you also use in the dropdown
- Maybe get rid of the subtitle? It's often truncated and doesn't seem to add much value
- Remember searched terms and show them in a section on the top. You can filter them locally. This gives another speed boost for frequently searched terms. I do something similar with the [GitHub Repository Search](https://www.raycast.com/thomas/github-repository-search).

apple-developer-docs 2022-07-29 at 09 16 54
 
Wonderful first contribution and I'm sure it's something that our team will adopt. Also, feel free to post about the extension in our [Slack Community](https://raycast.com/community).

File: extensions/apple-developer-docs/package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  "icon": "apple-developer-docs.png",
+  "author": "cali",
+  "categories": [
+    "Developer Tools"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search",
```
Comment (thomaspaulmann) [inline extensions/apple-developer-docs/package.json:15]:
```suggestion
      "title": "Search Documentation",
      "subtitle": "Apple Developer",
```

**Pull request merged**